server/script/migrateAssetsLocation.js

```diff
@@ -0,0 +1,194 @@
+'use strict';
+
+const {
+  Activity,
+  ContentElement,
+  Revision,
+  sequelize
+} = require('../shared/database');
+const {
+  getActivityMetadata,
+  getElementMetadata,
+  SCHEMAS
+} = require('../../config/shared/activities');
+const flatten = require('lodash/flatten');
+const fromPairs = require('lodash/fromPairs');
+const get = require('lodash/get');
+const { Op } = require('sequelize');
+const Promise = require('bluebird');
+const { protocol } = require('../../config/server/storage');
+const storage = require('../shared/storage');
+const toPairs = require('lodash/toPairs');
+
+const regex = /repository\/assets\/(.*)/;
+const ceTypes = ['IMAGE', 'VIDEO', 'AUDIO', 'PDF', 'CAROUSEL', 'MODAL'];
+const revisionsTypes = ['CONTENT_ELEMENT', 'ACTIVITY'];
+const isFunction = fn => fn && typeof fn === 'function';
+const schemasIds = SCHEMAS.map(it => it.id);
+
+const mapTypeToAction = {
+  IMAGE: imageMigrationHandler,
+  CAROUSEL: embedsMigrationHandler,
+  MODAL: embedsMigrationHandler,
+  DEFAULT: defaultMigrationHandler
+};
+
+const mapEntityToAction = {
+  CONTENT_ELEMENT: migrateContentElement,
+  ACTIVITY: migrateActivity
+};
+
+const invokeAction = type => (...args) => {
+  const action = mapTypeToAction[type];
+  const defaultAction = mapTypeToAction.DEFAULT;
+  return isFunction(action) ? action(...args) : defaultAction(...args);
+};
+
+migrate()
+  .then(() => {
+    console.info('Migration script was executed successfully.');
+    process.exit(0);
+  })
+  .catch(error => {
+    console.error(error.message);
+    process.exit(1);
+  });
+
+async function migrate() {
+  const transaction = await sequelize.transaction();
+  await migrateActivities(transaction);
+  await migrateContentElements(transaction);
+  await migrateRevisions(transaction);
+  return transaction.commit();
+}
+
+async function migrateActivities(transaction) {
+  const activities = await Activity.findAll({ transaction });
+  return Promise.each(activities, async it => {
+    const payload = await migrateActivity(it);
+    return it.update(payload, { transaction });
+  });
+}
+
+async function migrateActivity(activity) {
+  const { repositoryId, data: meta } = activity;
+  const fileMetas = getActivityMetadata(activity)
+    .filter(it => it.type === 'FILE')
+    .map(it => it.key);
```

Comment:
This could be solved as one `reduce`. But for me this is still fine, since the number of iterations is negligible. 
This also applies on other places in this file where you use similar chain.